pkg/registry/generic/etcd/etcd_test.go

```diff
@@ -227,29 +176,30 @@ func TestEtcdList(t *testing.T) {
 		if item.context != nil {
 			ctx = item.context
 		}
-		fakeClient, registry := NewTestGenericEtcdRegistry(t)
-		if name, ok := item.m.MatchesSingle(); ok {
-			if key, err := registry.KeyFunc(ctx, name); err == nil {
-				key = etcdtest.AddPrefix(key)
-				fakeClient.Data[key] = item.in
-			} else {
-				key := registry.KeyRootFunc(ctx)
-				key = etcdtest.AddPrefix(key)
-				fakeClient.Data[key] = item.in
+		server, registry := NewTestGenericEtcdRegistry(t)
+
+		if item.in != nil {
+			if err := etcdstorage.CreatePodList(t, "/pods", registry.Storage, item.in); err != nil {
+				t.Errorf("Unexpected error %v", err)
 			}
-		} else {
-			key := registry.KeyRootFunc(ctx)
-			key = etcdtest.AddPrefix(key)
-			fakeClient.Data[key] = item.in
 		}
+
 		list, err := registry.ListPredicate(ctx, item.m, nil)
-		if e, a := item.succeed, err == nil; e != a {
-			t.Errorf("%v: expected %v, got %v: %v", name, e, a, err)
+		if err != nil {
+			t.Errorf("Unexpected error %v", err)
 			continue
 		}
-		if e, a := item.out, list; !api.Semantic.DeepDerivative(e, a) {
-			t.Errorf("%v: Expected %#v, got %#v", name, e, a)
+		if item.out != nil {
+			if e, a := item.out, list; !api.Semantic.DeepDerivative(e, a) {
+				t.Errorf("%v: Expected %#v, got %#v", name, e, a)
+			}
+		} else {
+			// verify what we obtained was what we set.
```

Comment:
As I wrote above - this is strange. I would prefer always having out != nil and now introducing this "else" here.